src/context.rs

```diff
@@ -55,6 +58,11 @@ pub trait CollectionContext: Send + Sync {
     async fn query(&self, query: QueryParams) -> Result<DataFrame, ODataError>;
 
     fn on_unsupported_feature(&self) -> OnUnsupported;
+
+    /// Validates the record batches that retunred from datafusion before encode them to xml
+    async fn validate(&self, _record_batches: &[RecordBatch]) -> Result<(), ODataError> {
```

Comment:
Can you explain the motivation behind this?

Your code is the one executing `query()` to return the original `DataFrame` from which these records are produced, so it's surprising to me that you'd want to validate your own result.

What type of validation are you running that cannot be done on the `DataFrame` creation stage? Is it based on schema, or the actual values?

Reply:
Hey @sergiimk,

Basically, we're trying to validate the actual values. We're using `DataFusion` with [federation](https://github.com/datafusion-contrib/datafusion-federation) (DataFusion => remote DBs) and attempting to perform complex join clauses during the `DataFrame` creation stage. We need a way to validate the values coming from the query to ensure that we have the correct data.

Let me know if you have any suggestions for this. We’re looking for access to the `RecordBatch` before generating the XML result so we can perform this validation.